extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
+| storage:object_path           | string    | The object_path for the asset, used along with bucket |
+| storage:requester_pays        | bool      | Is the data requester pays or is it data manager/cloud provider pays |
```

Comment:
requester_pays as a term is AWS specific, but I think it's also become the standard way of describing this

Reply:
Yeah, since Google uses this as well, and it's also a very descriptive term I think it's good to use generally.  

If Azure implements a similar mechanism my guess is they will adopt the same term

Reply:
> If Azure implements a similar mechanism my guess is they will adopt the same term
they were the only one of the six cloud providers I investigated to buck the trend and use `container` instead of `bucket`...

Reply:
oh, interesting -- I only use AWS, so I just assumed that other services didn't use bucket out of spite -- I'm fine with it if that's the common term.